orchagent/nexthopkey.h

```diff
@@ -15,9 +17,24 @@ struct NextHopKey
     string              alias;          // incoming interface alias
     uint32_t            vni;            // Encap VNI overlay nexthop
     MacAddress          mac_address;    // Overlay Nexthop MAC.
+    LabelStack          label_stack;    // MPLS label stack
 
     NextHopKey() = default;
-    NextHopKey(const std::string &ipstr, const std::string &alias) : ip_address(ipstr), alias(alias), vni(0), mac_address() {}
+    NextHopKey(const std::string &str, const std::string &alias) : alias(alias), vni(0), mac_address()
+    {
+        std::size_t label_delimiter = str.find(LABELSTACK_DELIMITER);
+        std::string ip_str;
+        if (label_delimiter != std::string::npos)
+        {
+            label_stack = LabelStack(str.substr(0, label_delimiter));
+            ip_str = str.substr(label_delimiter+1);
+        }
+        else
+        {
+            ip_str = str;
+        }
+        ip_address = ip_str;
+    }
```

Comment:
write a helper function to reduce code duplication.

Reply:
Done